src/transform-img/test-transform-img.ts

```diff
@@ -0,0 +1,240 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {prepareImageAnimation} from './transform-img.js';
+import {imgLoadPromise} from '../testing/utils.js';
+import {
+  setup as setupAnimations,
+  tearDown as tearDownAnimations,
+  offset,
+} from '../testing/animation-test-controller';
+
+const {expect} = chai;
+const threeByFourUri = 'data:image/gif;base64,R0lGODdhAwAEAIAAAP///////ywAAAAAAwAEAAACA4SPVgA7';
+
+describe('prepareImageAnimation', () => {
+  let testContainer;
+  let smallerImg;
+  let largerImg;
+  let transitionContainer;
+  let cleanupAnimation;
+  const curve = {x1: 0.8, y1: 0, x2: 0.2, y2: 1};
+  const head = document.head!;
+  const styles = {
+    'animationFillMode': 'forwards',
+    'animationDuration': '1000ms',
+  };
+
+  function getReplacement() {
+    return transitionContainer.firstChild.firstChild;
+  }
+
+  function startAnimation(srcImg, targetImg) {
+    const animation = prepareImageAnimation({
+      transitionContainer,
+      styleContainer: head,
+      srcImg,
+      targetImg,
+      styles,
+      curve,
+    });
+    animation.applyAnimation();
+    cleanupAnimation = animation.cleanupAnimation;
+  }
+
+  async function updateImg(img, fit, src) {
+    img.style.objectFit = fit;
+    img.src = src;
+    await imgLoadPromise(img);
+  }
+
+  before(() => {
+    setupAnimations();
+  });
+
+  after(() => {
+    tearDownAnimations();
+  });
+
+  beforeEach(() => {
+    cleanupAnimation = () => {};
+    smallerImg = document.createElement('img');
+    smallerImg.style.position = 'fixed';
+    largerImg = document.createElement('img');
+    largerImg.style.position = 'fixed';
+    testContainer = document.createElement('div');
+    transitionContainer = document.createElement('div');
+    testContainer.appendChild(smallerImg);
+    testContainer.appendChild(largerImg);
+    document.body.appendChild(testContainer);  
+    document.body.appendChild(transitionContainer);  
+  });
+
+  afterEach(() => {
+    cleanupAnimation();
+    document.body.removeChild(testContainer);
+  });
+
+  describe('smaller to larger', () => {
+    beforeEach(async () => {
+      await updateImg(smallerImg, 'contain', threeByFourUri);
+      smallerImg.style.width = '12px';
+      smallerImg.style.height = '12px';
+      smallerImg.style.top = '500px';
+      smallerImg.style.left = '100px';
+      await updateImg(largerImg, 'contain', threeByFourUri);
+      largerImg.style.width = '24px';
+      largerImg.style.height = '32px';
+      largerImg.style.top = '10px';
+      largerImg.style.left = '10px';
+    });
+
+    it('should start with the correct size and position', () => {
+      startAnimation(smallerImg, largerImg);
+      offset(0);
+
+      const replacement = getReplacement();
+      const {top, left, width, height} = replacement.getBoundingClientRect();
+      expect(top).to.be.closeTo(500, 0.3);
```

Comment:
make 0.3 a const

Add a line before this explaining that `500` is the style.top for the smaller img. 

Ditto for the rest

Reply:
Done